packages/core/src/auth/credential-provider.ts

```diff
@@ -197,20 +233,56 @@ export class StringMomentoTokenProvider extends CredentialProviderBase {
   }
 
   getCacheEndpoint(): string {
-    return this.allEndpoints.cacheEndpoint;
+    return this.allEndpoints.cacheEndpoint.endpoint;
+  }
+
+  isCacheEndpointSecure(): boolean {
+    if (this.allEndpoints.cacheEndpoint.secureConnection === undefined) {
+      return true;
+    }
+    return this.allEndpoints.cacheEndpoint.secureConnection;
   }
 
   getControlEndpoint(): string {
-    return this.allEndpoints.controlEndpoint;
+    return this.allEndpoints.controlEndpoint.endpoint;
+  }
+
+  isControlEndpointSecure(): boolean {
+    if (this.allEndpoints.controlEndpoint.secureConnection === undefined) {
+      return true;
+    }
+    return this.allEndpoints.controlEndpoint.secureConnection;
   }
 
   getTokenEndpoint(): string {
-    return this.allEndpoints.tokenEndpoint;
+    return this.allEndpoints.tokenEndpoint.endpoint;
+  }
+
+  isTokenEndpointSecure(): boolean {
+    if (this.allEndpoints.tokenEndpoint.secureConnection === undefined) {
+      return true;
+    }
+    return this.allEndpoints.tokenEndpoint.secureConnection;
   }
 
   areEndpointsOverridden(): boolean {
     return this.endpointsOverridden;
   }
+
+  withMomentoLocal(): CredentialProvider {
+    const momentoLocalOverride = {
+      endpoint: '127.0.0.1:8080',
```

Comment:
this is hardcoded for now but can be made configurable later when it's configurable on momento-local